PubNub/Core/PubNub+APNS.m

```diff
@@ -398,61 +209,69 @@ - (void)removeAllPushNotificationsFromDeviceWithPushToken:(id)pushToken
-                               pushType:(PNPushType)pushType
-                            environment:(PNAPNSEnvironment)environment
-                                  topic:(NSString *)topic
-                          andCompletion:(PNPushNotificationsStateModificationCompletionBlock)block {
-
-    PNRemoveAllPushNotificationsRequest *request = nil;
-    request = [PNRemoveAllPushNotificationsRequest requestWithDevicePushToken:pushToken
-                                                                     pushType:pushType];
+                                                 pushType:(PNPushType)pushType
+                                              environment:(PNAPNSEnvironment)environment
+                                                    topic:(NSString *)topic
+                                            andCompletion:(PNPushNotificationsStateModificationCompletionBlock)block {
+
+    PNPushNotificationManageRequest *request = nil;
+    request = [PNPushNotificationManageRequest requestToRemoveDeviceWithToken:pushToken pushType:pushType];
     
     if (pushType == PNAPNS2Push) {
         request.environment = environment;
         request.topic = topic;
     }
     
-    [self removeAllPushNotificationsWithRequest:request completion:block];
+    [self managePushNotificationWithRequest:request completion:block];
 }
 
-- (void)removeAllPushNotificationsWithRequest:(PNRemoveAllPushNotificationsRequest *)request
-                             completion:(PNPushNotificationsStateModificationCompletionBlock)block {
-    
-    PNLogAPICall(self.logger, @"<PubNub::API> Disable push notifications for device '%@'%@.",
-                 request.pushToken,
-                 request.pushType == PNAPNS2Push ? [NSString stringWithFormat:@" ('%@' topic in %@ environment)",
-                                                    request.topic,
-                                                    request.environment == PNAPNSDevelopment ? @"development" : @"production"]
-                                                 : @"");
-    
-    __weak __typeof(self) weakSelf = self;
-    
-    [self performRequest:request withCompletion:^(PNAcknowledgmentStatus *status) {
-        if (status.isError) {
-            status.retryBlock = ^{
-                [weakSelf removeAllPushNotificationsWithRequest:request completion:block];
+
+#pragma mark - Push notifications state audit
+
+- (void)fetchPushNotificationWithRequest:(PNPushNotificationFetchRequest *)userRequest
+                              completion:(PNPushNotificationsStateAuditCompletionBlock)handlerBlock {
+    PNOperationDataParser *responseParser = [self parserWithResult:[PNAPNSEnabledChannelsResult class]
+                                                            status:[PNErrorStatus class]];
+    PNPushNotificationsStateAuditCompletionBlock block = [handlerBlock copy];
+    PNParsedRequestCompletionBlock handler;
+
+#ifndef PUBNUB_DISABLE_LOGGER
```

Comment:
We could consider checking against the `PUBNUB_DISABLE_LOGGER` flag as a part of `PNLogAPICall`, so we would have such a check in one place only. Another idea could be overriding `debugDescription` for objects like `userRequest`. These two should shorten the overall syntax for logging. Let's leave it for now and consider pros and cons in the future